### PR TITLE
Add store_csv_line method for File

### LINKED ISSUE #23656: Missing write_csv_line()
Godot 3.0.6
Godot 3.1 alpha2

`File` has `get_csv_line()`, but there is no easy way to write a CSV line.
This is not just writing raw values separated by commas, because it needs potential escaping and quoting, so a corresponding helper would be useful to have.

### PR DESCRIPTION
**Pull request by Kanabenki**
Closes #23656 

I also edited the parameter of "get_csv_line" to be consistent with the other methods.

### CODE REVIEW

File: doc/classes/File.xml [hunk 1]
```diff
@@ -327,6 +327,17 @@
...
 		</method>
+		<method name="store_csv_line">
+			<return type="void">
+			</return>
+			<argument index="0" name="values" type="PoolStringArray" default="&quot;,&quot;">
+			</argument>
+			<argument index="1" name="delim" type="String" default="&quot;,&quot;">
+			</argument>
+			<description>
+				Store the given [PoolByteArray] in the file as a line formated in the CSV (Comma Separated Values) format. You can pass a different delimiter to use other than the default "," (comma).
```
Comment (Naryosha) [inline doc/classes/File.xml]: `PoolByteArray` -> `PoolStringArray`
formated -> formatted

I'd also add that the delimiter must be 1 char long.

File: core/os/file_access.h [hunk 2]
```diff
@@ -112,7 +112,8 @@ class FileAccess {
 	virtual int get_buffer(uint8_t *p_dst, int p_length) const; ///< get an array of bytes
 	virtual String get_line() const;
 	virtual String get_token() const;
-	virtual Vector<String> get_csv_line(String delim = ",") const;
+	virtual Vector<String> get_csv_line(const String &p_delim = ",") const;
+	virtual void store_csv_line(const Vector<String> &p_values, const String &p_delim = ",");
```
Comment (Naryosha) [inline core/os/file_access.h]: I'd suggest grouping this method with the other `store*`'s below.

File: core/bind/core_bind.cpp [hunk 3]
```diff
@@ -1944,6 +1949,7 @@ void _File::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_error"), &_File::get_error);
 	ClassDB::bind_method(D_METHOD("get_var"), &_File::get_var);
 	ClassDB::bind_method(D_METHOD("get_csv_line", "delim"), &_File::get_csv_line, DEFVAL(","));
+	ClassDB::bind_method(D_METHOD("store_csv_line", "values", "delim"), &_File::store_csv_line, DEFVAL(","));
```
Comment (Naryosha) [inline core/bind/core_bind.cpp]: I'd suggest grouping this method with the other `store*`'s below.

**Review by ghost — COMMENTED** [review]
Thanks for contributing!

**Comment by Kanabenki** [comment]
Thanks for the review ! I've implemented the changes you suggested.

File: core/os/file_access.cpp [hunk 4]
```diff
@@ -525,6 +525,28 @@ void FileAccess::store_line(const String &p_line) {
...
+void FileAccess::store_csv_line(const Vector<String> &p_values, const String &p_delim) {
+
+	ERR_FAIL_COND(p_delim.length() != 1);
+
+	String line = "";
+	int size = p_values.size();
+	for (int i = 0; i < size; ++i) {
+		String value = p_values[i];
+
+		if (value.find("\"") != -1 || value.find(p_delim) != -1) {
```
Comment (Zylann) [inline core/os/file_access.cpp]: I think quotes are also necessary when there are newlines in the value, I found LibreOffice Calc does that

Comment (Kanabenki) [reply core/os/file_access.cpp]: Thanks, I added it.

**Pull request merged**